.github/workflows/autograding.yml

```diff
@@ -0,0 +1,71 @@
+# This is a basic workflow to help you get started with Actions
+
+name: Autograding
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push or pull request events but only for the "main" branch
+  push:
+    branches: [ "main", "master" ]
+  pull_request:
+    branches: [ "main", "master" ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - name: Student Repo
+        uses: actions/checkout@v3
+        timeout-minutes: 10
+        with:
```

Comment:
For each step 10 minuts is too much. 
10 mins seemed to be a lot for the whole build itself.